### PR TITLE
Template Color updates

### PR DESCRIPTION
**Pull request by davidortinau**
I propose not using Light/Dark for color names we provide, and instead using Primary, Secondary, Tertiary. Let's leave Light/Dark to AppThemeBinding and if the dev wants to add semantic color names.

To replace the grays, I propose using names that indicate hue. This also matches the Fluent color guideline.

### CODE REVIEW

**Comment by Eilon** [comment]
The colors in practice seem quite different from what they were before. Can you show screenshot before/after for this? I'm wondering whether the contrast is still sufficient for accessibility needs.

Example: Where it's consumed, `DarkGray` was replaced with `Gray950`, but the actual color changed from `#505050` to `#141414`, which is significantly darker.

Before:
lighter dark color

After:
darker dark color

**Review by Redth — APPROVED** [review]

**Comment by davidortinau** [comment]
@Eilon will do.

File: src/Templates/src/templates/maui-mobile/Resources/Styles/Styles.xaml [hunk 1]
```diff
@@ -345,15 +345,15 @@
 
     <Style TargetType="Shell" ApplyToDerivedTypes="True">
         <Setter Property="Shell.BackgroundColor" Value="{AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource Black}}" />
-        <Setter Property="Shell.ForegroundColor" Value="{AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource White}}" />
-        <Setter Property="Shell.TitleColor" Value="{AppThemeBinding Light={StaticResource Gray200}, Dark={StaticResource White}}" />
+        <Setter Property="Shell.ForegroundColor" Value="{OnPlatform WinUI={StaticResource Primary}, Default={StaticResource White}}"></Setter>
```
Comment (Redth) [inline src/Templates/src/templates/maui-mobile/Resources/Styles/Styles.xaml:348]:
```suggestion
        <Setter Property="Shell.ForegroundColor" Value="{OnPlatform WinUI={StaticResource Primary}, Default={StaticResource White}}" />
```

Just a nit on introducing the closing tag

**Comment by Eilon** [comment]
Nifty, thank you for checking this!

**Review by Redth — APPROVED** [review]

**Review by mattleibow — APPROVED** [review]

**Review by rachelkang — APPROVED** [review]

**Pull request merged**